north/src/runtime/mod.rs

```diff
@@ -113,6 +113,22 @@ pub async fn run(config: &Config) -> Result<()> {
         state.applications.len()
     );
 
+    // handle SIGINT and SIGTERM
+    let tx_interrupt = tx.clone();
+    ctrlc::set_handler(move || {
```

Comment:
Why not moving a copy of the tx handle of the main event loop to the handler? This would be way simpler.

Reply:
this is exactly what this is doing, isn't it?